pyomo/contrib/mindtpy/util.py

```diff
@@ -15,6 +15,8 @@
 from pyomo.opt.results import ProblemSense
 from pyomo.solvers.plugins.solvers.persistent_solver import PersistentSolver
 
+MAX_SYMBOLIC_DERIV_SIZE = 1000
```

Comment:
We are not using this limit anywhere (this was from GDPopt). I would remove it since it sounds pretty arbitrary.

Reply:
It is arbitrary. The original implementation of differentiate used an interface to CasADi, which no longer exists. From there, we moved to sympy, and then @michaelbynum created the AD we have in core today.

The default to sympy in GDPopt remained to maintain consistency, but for larger models, we found Michael's implementation much faster, hence the arbitrary switch. GDPopt can likely be switched over to use reverse_symbolic by default as well.